node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -127,17 +127,18 @@ impl<C: Context> ActiveValidator<C> {
     pub(crate) fn on_new_vote<R: Rng + CryptoRng + ?Sized>(
         &mut self,
         vhash: &C::Hash,
-        timestamp: Timestamp,
+        mut timestamp: Timestamp,
         state: &State<C>,
         instance_id: C::InstanceId,
         rng: &mut R,
     ) -> Vec<Effect<C>> {
         if let Some(evidence) = state.opt_evidence(self.vidx) {
             return vec![Effect::WeEquivocated(evidence.clone())];
         }
-        if self.earliest_vote_time(state) > timestamp {
-            warn!(%timestamp, "skipping outdated confirmation");
-        } else if self.should_send_confirmation(vhash, timestamp, state) {
+        // TODO: `timestamp` should be the _current_ timestamp and we need to delay incoming votes
+        // with a future timestamp.
+        timestamp = self.earliest_vote_time(state).max(timestamp);
+        if self.should_send_confirmation(vhash, timestamp, state) {
```

Comment:
In the body `should_send_confirmation`, `timestamp` is being compared to `vote(vhash).timestamp` – doesn't it mean that `timestamp` was being compared to itself?

Reply:
It does! It's supposed to be the current timestamp but we don't pass that in yet, so we just set it to the vote's… :confused: